modules/tensor_mechanics/src/kernels/GeneralizedPlaneStrainOffDiag.C

```diff
@@ -107,7 +108,7 @@ GeneralizedPlaneStrainOffDiag::computeOffDiagJacobianScalar(unsigned int jvar)
       computeDispOffDiagJacobianScalar(0, jvar);
     else if (_num_disp_var == 2 && _var.number() == _disp_var[1]->number())
       computeDispOffDiagJacobianScalar(1, jvar);
-    else if (isParamValid("temperature") ? _var.number() == _temp_var->number() : 0)
+    else if (_temp_var && _var.number() == _temp_var->number())
```

Comment:
I don't get this logic. If `_var.number() == _temp_var` then this shouldn't be off diagonal, or is this special thing with scalar vars?

Reply:
Separate instances of this class are set up for the each of the displacements and the temperature, each with a different `variable` parameter. `_var` is that field variable.  `jvar` is the scalar variable. Off-diagonal coupling terms are computed for the interaction between the field var `_var` and scalar var `jvar`. I think this is right.

Reply:
In looking at this a little more, the only reason we need `temperature` and `displacements` is to figure out what variable `_var` is. We should probably still pass those parameters in to this class, but we could sort all that out in the constructor and store off an enumeration or something that tells is whether we're working with disp_x, disp_y, or temperature. That might make this code easier to understand.

I can take a stab at refactoring this, but I think that should be separate from this PR. I think we need to do more testing on the Jacobians from this to make sure they're right while we're at it.